src/js/stepped-progress.js

```diff
@@ -1,24 +1,172 @@
+	 *
+	 * @access public
+	 * @returns {Array<SteppedProgressStep>} Returns an array of steps.
+	 */
+	getCompletedSteps() {
+		return this._steps.filter(step => step.isComplete());
+	}
+
+	/**
+	 * Get whether a step exists at a given index (0-based).
+	 *
+	 * @access public
+	 * @param {Number} index - The index to check.
+	 * @returns {Boolean} Returns whether a step exists at a given index.
+	 */
+	hasStepAtIndex(index) {
+		return Boolean(this._steps[index]);
+	}
+
+	/**
+	 * Get the step at a given index (0-based).
+	 *
+	 * @access public
+	 * @param {Number} index - The index of the step to get.
+	 * @returns {SteppedProgressStep} Returns the step at the given index.
+	 * @throws {Error} Will throw an error if there is no step at the given index. Use {@link SteppedProgress#hasStepAtIndex} to check.
+	 */
+	getStepAtIndex(index) {
+		if (!this.hasStepAtIndex(index)) {
+			throw new Error(`No step at index: ${index}`);
+		}
+		return this._steps[index];
+	}
+
+	/**
+	 * Get the step which has the "current" state. If there are multiple steps with this state then
+	 * the last one will be returned.
+	 *
+	 * @access public
+	 * @returns {SteppedProgressStep} Returns the current step.
+	 */
+	getCurrentStep() {
+		return this._steps.filter(step => step.isCurrent()).pop();
+	}
+
+	/**
+	 * Get the last step in the stepped progress.
+	 *
+	 * @access public
+	 * @returns {SteppedProgressStep} Returns the last step.
+	 */
+	getLastStep() {
+		return this._steps[this._steps.length - 1];
+	}
+
+	/**
+	 * Get whether all steps have the "completed" state.
+	 *
+	 * @access public
+	 * @returns {Boolean} Returns whether all steps are completed.
+	 */
+	isComplete() {
+		return (this.getCompletedSteps().length === this._steps.length);
+	}
+
+	/**
+	 * Get the next future step (a step which does not have the "current", "complete", or "error"
+	 * states). If no such step exists, the last step will be returned.
+	 *
+	 * @access public
+	 * @returns {SteppedProgressStep} Returns the next step.
+	 */
+	getNextStep() {
+		if (!this.isComplete()) {
+			return this._steps.find(step => step.isFuture()) || this.getLastStep();
+		}
+		return this.getLastStep();
+	}
+
+	/**
+	 * Mark the current step as "complete" and then mark the next step as "current". If all steps
+	 * have the "complete" state then this method does nothing.
+	 *
+	 * @access public
+	 * @returns {void}
+	 */
+	progress() {
+		if (!this.isComplete()) {
+			const currentStep = this.getCurrentStep();
+			if (currentStep) {
+				currentStep.markAsComplete();
+			}
+		}
+		if (!this.isComplete()) {
+			this.getNextStep().markAsCurrent();
+		}
+	}
+
+	/**
+	 * Construct step instances and store them on the  `_steps` and `_stepElementInstanceMap` properties.
+	 *
+	 * @access private
+	 * @returns {void}
+	 */
+	_constructSteps() {
+		const elements = this.steppedProgressElement.querySelectorAll(`.${classNames.step}`);
+		this._stepElementInstanceMap = new Map([...elements].map(element => {
```

Comment:
`_stepElementInstanceMap` looks to only be used in this method, is the reason to have this map attached to the instance to make removing the `SteppedProgressStep` instances simpler?

Reply:
Ah yes, I intended to use this to make adding/removing steps easier but then never used it. Perhaps it's better to leave it out for now

Reply:
I've removed this for now, we can re-add later if we need a quick way to get the step instance that corresponds to an element